src/mps/projmposum.jl

```diff
@@ -109,10 +85,11 @@ The MPS `psi` must have compatible bond indices with
 the previous projected MPO tensors for this
 operation to succeed.
 """
-function position!(P::ProjMPOSum, psi::MPS, pos::Int)
-  for M in P.pm
-    position!(M, psi, pos)
+function position!(A::AbstractSum, psi::MPS, pos::Int)
+  for t in terms(A)
+    position!(t, psi, pos)
   end
+  return A
```

Comment:
Better to use:
```julia
  new_terms = map(term -> position!(term, psi, pos), terms(A))
  return set_terms(A, new_terms)
```
This is more general since some `position!` functions don't modify in-place, but instead return new environments (such as in the distributed case).

This is how it is written in https://github.com/ITensor/ITensorParallel.jl/pull/15.

Reply:
I see, great. And in the new code is the use of an in-place `position!` justified in the sense that if it does do an in-place modification, then `new_terms` will just contain a bunch of new references, but still to the old, now modified, objects?

Reply:
Yeah, that's right. Unfortunately it could be slightly breaking since someone could have written a `position!` overload that doesn't return the modified `term` but hopefully they will reach out if they actually have that issue.

Really we should be using `position!!` following the [BangBang.jl](https://juliafolds.github.io/BangBang.jl/stable/) convention, which we use for functions like `contract!!` and `permutedims!!`, which represents that the function either is in-place or allocates the output, but let's just keep it as `position!` for now to not overcomplicate this PR.